core/src/main/java/io/questdb/griffin/engine/groupby/DistinctTimeSeriesRecordCursorFactory.java

```diff
@@ -77,7 +77,13 @@ public RecordCursorFactory getBaseFactory() {
 
     @Override
     public RecordCursor getCursor(SqlExecutionContext executionContext) throws SqlException {
-        return cursor.of(base.getCursor(executionContext), executionContext);
+        final RecordCursor baseCursor = base.getCursor(executionContext);
```

Comment:
the constructor of this class is overlooked, if map was to OOM we will leak things